### DISCUSSION TITLE
Equivalent of `yq --collect --tojson`?

### DISCUSSION THREAD

**Discussion by ghostsquad** [TomWright/dasel#128] [General]
Given a yaml file that has multiple documents, e.g.

```
foo: bar
---
baz: biz
```

I'd like to get an _array_ of each document. Is this possible with dasel?

**Comment by TomWright** [comment]
This isn't possible right now but is a fairly simple use-case.

I'll add a `--merge-input-documents` flag that can do this. E.g.

```bash
echo 'foo: bar
---
baz: biz' | dasel -r yaml -w json --merge-input-documents .
[
  {
    "foo": "bar"
  },
  {
    "baz": "biz"
  }
]

```

The work is done on the `merge-documents` branch which you can install from with the following:
```bash
go install github.com/tomwright/dasel/cmd/dasel@merge-documents
```

The documentation is here: https://app.gitbook.com/@tomwright/s/dasel/usage/flags/merge-input-documents

**Reply by link4531** [reply]
Am I correct in assuming this feature could also merge two separate files (documents?) with similar structure?

file1: `foo: bar` 
file2: `baz: biz`
**this is a terrible example**
`cat file1 file2 | dasel -yaml -json --merge-input-documents`

Sorry if I am asking the same question as mentioned above. 

Also any chance we  could get a Release built with this feature included maybe tagged as 'preview' or 'beta'. 
I do not have the ability to utilize the 'go install ...' per your instruction.

**Reply by TomWright** [reply]
Assuming your documents are correctly formatted and that the `cat` output is properly formed, yes.

x.yaml
```yaml
foo: bar
---

```

y.yaml
```yaml
baz: biz

```

Without flag:
```bash
$ cat x.yaml y.yaml| ~/go/bin/dasel -p yaml .         
foo: bar
---
baz: biz

```

With flag:
```bash
$ cat x.yaml y.yaml| ~/go/bin/dasel --merge-input-documents -p yaml .
- foo: bar
- baz: biz

```

This flag tells dasel to merge all the given documents into an array/list, where `.[0]` contains the first document, `.[1]` contains the second document and so on.

---

I will merge and release this feature once I know it's named appropriately and doesn't have any bugs (more tests).

**Reply by link4531** [reply]
Super stoked for this feature. 

Question is there a priority to the merge order?
such as one file contains foo: bar and a later file in the array contains foo: biz << I assume the last value to be read will be written in the output?

**Reply by TomWright** [reply]
That's great to hear! I'm busy at work at the moment but will set aside some time to get this done soon 👍 

Dasel just reads and merges the document in the order they are provided, so yes your assumption is correct.

Do you have some basic examples of what you may try to do with this feature? They will give me some good use-cases to test against.

**Reply by TomWright** [reply]
This has been released under `v1.15.0`.

**Comment by TomWright** [comment]
@link4531 

Take note that merge in this case means taking all of the input documents and adding them to a single array of those documents.

Input: `a`, `b`, `c`
Output: `[a, b, c]`

Where:
a: `{"number": 1}`
b: `{"number": 2}`
c: `{"number": 3}`

Output:
```json
[
    {"number": 1},
    {"number": 2},
    {"number": 3}
]
```

The use of "merge" here could be mistaken in that you may expect the following output:
```json
{
    "number": 3
}
```

I'm considering adding a new feature that will provide the second output but am not yet sure what the usage will look like.

https://github.com/TomWright/dasel/issues/133
